core/modules/util/ThreadPool.cc

```diff
@@ -204,6 +220,21 @@ bool ThreadPool::release(PoolEventThread *thrd) {
 /// Change the size of the thread pool.
 void ThreadPool::resize(unsigned int targetThrdCount) {
     {
+        LOGS(_log, LOG_LVL_WARN, "&&&WARN ThreadPool::resize " << targetThrdCount);
+        {
+            std::lock_guard<std::mutex> lockPool(_mxPool);
+            /// This is not expected to be used in cases where low CPU usage threads
+            /// are removed from the pool. In those cases, the targetThrdCount is expected to be
+            /// less than 100 while _maxThreadCount would be several thousand.
+            if (targetThrdCount >= _maxThreadCount) {
+                LOGS(_log, LOG_LVL_WARN, "ThreadPool::resize target count " << targetThrdCount
+                        << " is >= than max " << _maxThreadCount);
+                // Need at least one thread available to deal with released threads.
+                _maxThreadCount = targetThrdCount + 1;
+                _cvPool.notify_all();
```

Comment:
Why notifying all threads instead `notify_one` if increasing the thread count by 1?